openpype/plugins/load/add_site.py

```diff
@@ -12,21 +22,60 @@ class AddSyncSite(load.LoaderPlugin):
     icon = "download"
     color = "#999999"
 
+    _sync_server = None
+
+    @property
+    def sync_server(self):
+        if not self._sync_server:
+            manager = ModulesManager()
+            self._sync_server = manager.modules_by_name["sync_server"]
+
+        return self._sync_server
+
     def load(self, context, name=None, namespace=None, data=None):
         self.log.info("Adding {} to representation: {}".format(
             data["site_name"], data["_id"]))
-        self.add_site_to_representation(data["project_name"],
-                                        data["_id"],
-                                        data["site_name"])
+        family = context["representation"]["context"]["family"]
+        project_name = data["project_name"]
+        repre_id = data["_id"]
+        self.add_site_to_representation(project_name,
+                                        repre_id,
+                                        data["site_name"],
+                                        is_main=True)
+
+        if family == "workfile":
+            links = get_linked_ids_for_representations(project_name,
+                                                       [repre_id],
+                                                       link_type="reference")
+            for link_repre_id in links:
+                self.add_site_to_representation(project_name,
+                                                link_repre_id,
+                                                data["site_name"],
+                                                is_main=False)
+
         self.log.debug("Site added.")
 
-    @staticmethod
-    def add_site_to_representation(project_name, representation_id, site_name):
-        """Adds new site to representation_id, resets if exists"""
-        manager = ModulesManager()
-        sync_server = manager.modules_by_name["sync_server"]
-        sync_server.add_site(project_name, representation_id, site_name,
-                             force=True)
+    def add_site_to_representation(self, project_name, representation_id,
+                                   site_name, is_main):
+        """Adds new site to representation_id, resets if exists
+
+        Args:
+            project_name (str)
+            representation_id (ObjectId):
+            site_name (str)
+            is_main (bool): true for main representation, false for referenced
+                loaded repres. Drives if site state should be reset.
+                (it should be for main, not for referenced as they might be
+                shared from multiple workfiles). In necessary cases, referenced
+                repres should be reset (re-downloaded) manually.
```

Comment:
Thanks - it is more readable. But since I don't have too much knowledge about site sync and this area of the code base I must admit that it's still rather hard to understand exactly what is happening.

In essence for a particular project's representation and a given site name this would make sure all files are set up to sync at the given site.

When `force` is set (for `is_main=True`) then the files will be redownloaded even if they already existed locally?
When `force_only_broken` is set, then what happens? Does it only redownload if the file locally has an invalid checksum, or what is considered "broken"?